commit c62262a154ef436d9adbc98cffde93034060d602
Author: Mark Thomas <markt@apache.org>
Date:   Wed Jun 3 13:18:32 2015 +0000

    inBuffer needs to be volatile for double checked locking to work.
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@1683322 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/org/apache/coyote/http2/Stream.java b/java/org/apache/coyote/http2/Stream.java
index 9a0c4d5..2734e56 100644
--- a/java/org/apache/coyote/http2/Stream.java
+++ b/java/org/apache/coyote/http2/Stream.java
@@ -367,7 +367,7 @@ public class Stream extends AbstractStream implements HeaderEmitter {
         private byte[] outBuffer;
         // This buffer is the destination for incoming data. It is normally is
         // 'write mode'.
-        private ByteBuffer inBuffer;
+        private volatile ByteBuffer inBuffer;
 
         private boolean endOfStream = false;
 
